lib/sm/client.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+require 'faraday'
+require 'multi_json'
+require 'common/client/errors'
+require 'sm/client_session'
+require 'sm/configuration'
+require 'sm/parser'
+require 'sm/api/sessions'
+require 'sm/api/triage_teams'
+require 'sm/api/folders'
+require 'sm/api/messages'
+
+module SM
+  class Client
+    include SM::API::Sessions
+    include SM::API::TriageTeams
+    include SM::API::Folders
+    include SM::API::Messages
+
+    REQUEST_TYPES = %i(get post delete).freeze
+    USER_AGENT = 'Vets.gov Agent'
+    BASE_REQUEST_HEADERS = {
+      'Accept' => 'application/json',
+      'Content-Type' => 'application/json',
+      'User-Agent' => USER_AGENT
+    }.freeze
+
+    attr_reader :config, :session
+
+    def initialize(config:, session:)
+      @config = config.is_a?(Hash) ? SM::Configuration.new(config) : config
+      @session = session.is_a?(Hash) ? SM::ClientSession.new(session) : session
+
+      raise ArgumentError, 'config is invalid' unless @config.is_a?(Configuration)
+      raise ArgumentError, 'session is invalid' unless @session.valid?
+    end
+
+    def authenticate
+      @session = get_session
+    end
+
+    private
+
+    def perform(method, path, params = nil, headers = nil)
```

Comment:
Personal nit, but I prefer named args in cases like this where `params` is frequently nil and `headers` isn't.